src/webots/gui/WbX3dStreamingServer.cpp

```diff
@@ -0,0 +1,256 @@
+  QByteArray reply;
+  if (mX3dWorldTextures.contains(requestedUrl))
+    reply = WbHttpReply::forgeImageReply(mX3dWorldTextures[requestedUrl]);
+  else
+    reply = WbHttpReply::forge404Reply();
+  socket->write(reply);
+}
+
+void WbX3dStreamingServer::processTextMessage(QString message) {
+  if (message.startsWith("x3d")) {
+    QWebSocket *client = qobject_cast<QWebSocket *>(sender());
+    WbLog::info(tr("Streaming server: Client set mode to: X3D."));
+    mPauseTimeout = message.endsWith(";broadcast") ? -1 : 0;
+    if (!WbWorld::instance()->isLoading())
+      startX3dStreaming(client);
+    // else streaming is started once the world loading is completed
+    return;
+  } else if (message == "reset") {
+    // reset nodes visibility
+    QWebSocket *client = qobject_cast<QWebSocket *>(sender());
+    foreach (WbBaseNode *node, WbWorld::instance()->viewpoint()->getInvisibleNodes())
+      client->sendTextMessage(QString("visibility:%1:1").arg(node->uniqueId()));
+    resetSimulation();
+    QString state = WbAnimationRecorder::instance()->computeUpdateData(true);
+    if (!state.isEmpty()) {
+      foreach (QWebSocket *client, mWebSocketClients)
+        sendWorldStateToClient(client, state);
+    }
+    sendToClients("reset finished");
+    return;
+  }
+  WbStreamingServer::processTextMessage(message);
+}
+
+void WbX3dStreamingServer::startX3dStreaming(QWebSocket *client) {
+  try {
+    if (WbWorld::instance()->isModified() || mX3dWorldGenerationTime != WbSimulationState::instance()->time())
+      generateX3dWorld();
+    sendWorldToClient(client);
+    // send the current simulation state to the newly connected client
+    const QString &stateMessage = simulationStateString();
+    if (!stateMessage.isEmpty())
+      client->sendTextMessage(stateMessage);
+    WbLog::info(
+      tr("Streaming server: New client [%1] (%2 connected client(s)).").arg(clientToId(client)).arg(mWebSocketClients.size()));
+  } catch (const QString &e) {
+    WbLog::error(tr("Streaming server: Cannot send world date to client [%1] because: %2.").arg(clientToId(client)).arg(e));
+  }
+}
+
+void WbX3dStreamingServer::sendUpdatePackageToClients() {
+  sendActivityPulse();
+
+  if (mWebSocketClients.size() > 0) {
+    qint64 currentTime = QDateTime::currentMSecsSinceEpoch();
+    if (mLastUpdateTime < 0.0 || currentTime - mLastUpdateTime >= 1000.0 / WbWorld::instance()->worldInfo()->fps()) {
+      QString state = WbAnimationRecorder::instance()->computeUpdateData(false);
+      if (!state.isEmpty()) {
+        foreach (QWebSocket *client, mWebSocketClients) {
+          sendWorldStateToClient(client, state);
+          pauseClientIfNeeded(client);
+        }
+      }
+      mLastUpdateTime = currentTime;
+    }
+  }
+}
+
+bool WbX3dStreamingServer::prepareWorld() {
+  try {
+    bool regenerationRequired = mX3dWorldReferenceFile != WbWorld::instance()->fileName() || mX3dWorldGenerationTime != 0.0;
```

Comment:
```suggestion
    const bool regenerationRequired = mX3dWorldReferenceFile != WbWorld::instance()->fileName() || mX3dWorldGenerationTime != 0.0;
```

Reply:
`regenerationRequired` is changed at line 146, so it cannot be changed to const.